src/simulation/hdf_simulation_store.jl

```diff
@@ -579,28 +584,48 @@ function _get_dataset(::Type{OptimizerStats}, store::HdfSimulationStore, problem
     return store.optimizer_stats_datasets[problem_name]
 end
 
-function _get_dataset(store::HdfSimulationStore, problem_name::Symbol)
-    return store.datasets[problem_name]
+function _get_dataset(store::HdfSimulationStore, model_name::Symbol)
+    return store.datasets[model_name]
 end
 
-function _get_dataset(store::HdfSimulationStore, model, opt_container_key::VariableKey)
-    return getfield(store.datasets[model], STORE_CONTAINER_VARIABLES)[opt_container_key]
+function _get_dataset(
+    store::HdfSimulationStore,
+    model_name::Symbol,
```

Comment:
`model_name` and `problem_name` appear to be used interchangeably. Also, why does OptimizationResultCacheKey have `model` instead of `problem`? I'm still not clear on these terms.

Reply:
Still working through the changes to make the naming consistent